mne/preprocessing/ica.py

```diff
@@ -1599,8 +1608,8 @@ def detect_artifacts(self, raw, start_find=None, stop_find=None,
                          ecg_ch=None, ecg_score_func='pearsonr',
                          ecg_criterion=0.1, eog_ch=None,
                          eog_score_func='pearsonr',
-                         eog_criterion=0.1, skew_criterion=-1,
-                         kurt_criterion=-1, var_criterion=0,
```

Comment:
Why do we change the default criteria here? This would change the behavior of `detect_artifacts` between versions without any warning.

Reply:
Well, my reasoning was that this way we at least actually _keep_ the effect of the default values, because before this PR, [0] would index the lowest score and [-1] the highest, and know it is inverted (i.e. now it is as it is (and used to be) documented).

But you are right, it would be probably good to introduce a warning. Could you point me to a template for a warning about changing the behavior of a function? After all the bug is only in the incompatibility between documentation and code, so someone who knew the code and did not care about the docstring could be surprised ...

Reply:
The least surprise from a user perspective would be to keep the old behavior and old defaults, and update the docstring to say what this does.

If our old defaults made no sense, we can have a deprecation cycle where we set them to `None` for a cycle in the signature, and in 0.18 `None` does what was done in 0.17 (with a warning that not setting it explicitly will change in 0.19) and then in 0.19 we change the default to something sane.

Reply:
I agree with @larsoner, BTW @jeythekey - you should better ask Eric about how to implement deprecations, I never did this so I would have to look around in our codebase, but Eric knows all this off the top of his head.

Reply:
So, in principle we would revert this PR and replace it with a warning, that in the next release the function will do what it used to say it would do in the docstring?
Is this really necessary in this case, where the docstring clearly states "Note. This is still experimental and will most likely change over the next releases."?
As it is now, at least the defaults will do the same as before (they were only changed, because of the bug-fix in the code), only when the criteria args used to be set contrary to what would be sensible given the docstring by a user, the behavior will change. So, this should only negatively affect those who dug through the code before, identified the bug and decided to silently adapt their usage instead of reporting the bug.

So, please, one last word by @larsoner: Should I really revert the fix for sorting the compos in descending order and replace the fix by a warning about a future fix?

Reply:
could you explain the sorting bug you mention? The whats new entry says "Fix descending sorting of scores for integer" - why do we want to change the sorting, why is descending not ok?

Reply:
The doc said "descending", but the code did "ascending". So `skew_criterion=0` returned the score with the smallest absolute skewness, contrary to what was further explained in the doc by "E.g. range(2) would return the two sources with the highest score." I find it intuitive as the docstring says, so I change the code to do just that. But of course I am open to change it to your liking! If you think we should stick to ascending sorting I would just change the doc.

Reply:
No, descending is good, it should be descending, so this is a bug fix. Then:
* change the what's new to state that ascending was changed into descending. I read the "Fix descending sorting" as if it was descending and was changed.
* I am ok with changing the criteria now, otherwise it would be nonsense. I somehow didn't before get that these "criteria" are indices. What do you think @larsoner? In case someone actually uses these we could set these to None and issue a warning.

Reply:
> In case someone actually uses these we could set these to None and issue a warning.

Even in this case if someone had previously passed values, now the behavior will be changed. I guess since the behavior is wrong and this is a bugfix, we can change the behavior like you say and make it clear in the BUG section that this has been fixed.

@agramfort are you okay with this?

Reply:
> Even in this case if someone had previously passed values ...

Actually, _only_ in this case ... (because the default values were inverted to retain the old behavior - and they make sense now, I guess, as they did before)

And what do you think about [this warning](https://github.com/mne-tools/mne-python/pull/6246/commits/1aacc1f59c0acfec0f3f7b5e93e53eb8695f82bf#diff-5a816b651c2f6b26c8899e97bfe6549aR2153) (at the end) I tried to implement (with the wrong warning class, apparently)?

Reply:
@larsoner and @mmagnuski should I then just remove the warning? I changed the bug section as you suggested. Because then we should be good to go, or is there anything else, @agramfort ?

Reply:
Yes let's remove the warning if it's a bug. Then wait for @agramfort to comment next week to make sure it's reasonable